### DISCUSSION TITLE
How to use with Expo Router?

### DISCUSSION THREAD

**Discussion by fredrikburmester** [callstack/react-native-bottom-tabs#21] [Q&A]
As the title says, how would I use this with Expo Router? How to work with layouts and pages? Thanks!

**Comment by okwasniewski** [comment]
You can find instructions here:

https://x.com/baconbrix/status/1843654478232531395?s=46&t=D61Tja_PFq4BVo_6xmK5bQ

Here is the link to Example Repo:

https://github.com/EvanBacon/expo-router-react-native-bottom-tabs

It should land in the docs soon

**Reply by fredrikburmester** [reply]
Hey, thanks for the quick answer, it seems to work somewhat (see image below), 

1. How do i fill the screen, so it goes up behind the status bar? 

![Simulator Screenshot - iPhone 16 Pro - 2024-10-08 at 21 01 26](https://github.com/user-attachments/assets/287416a2-a1e8-4f02-ad84-11d93596cee7)

2. I don't know if this is related but i get this type error and therefore no completions. 

![Screenshot 2024-10-08 at 20 43 16](https://github.com/user-attachments/assets/2dff6181-b6bb-4e2f-899e-ede2baab647a)

**Reply by okwasniewski** [reply]
Hey #23 this PR should fix types issues

**Comment by okwasniewski** [comment]
Are you using the latest version (0.0.7)? It should have correct typings. 

What's your structure to embed the top bar? Can you provide a minimal reproducible example? Probably we will need a prop like `ignoreSafeAreaInsets`

**Reply by fredrikburmester** [reply]
Yes i'm using 0.0.7.

No I don't have a MRE, could try to produce one later this week though if you still need it.

**Reply by fredrikburmester** [reply]
Ok so I used your example in the PR instad of Evans and now I have the types in the NativeTab, but still got the typescript error in vscode.

**Reply by fredrikburmester** [reply]
There are also some wrong types in the tabBarIcon:
![Screenshot 2024-10-08 at 22 09 54](https://github.com/user-attachments/assets/39d4e1c9-ae9b-4189-b550-b4cc309775b1)
